thoth/common/logging.py

```diff
@@ -149,6 +149,29 @@ def _get_sentry_integrations() -> List[object]:
     return integrations
 
 
+def before_send_handler(event, hint):
+    """Filter the errors caught before sending to Sentry.
+
+    This function ignores the exceptions passed in as a environment variable in a comma separated manner.
+    """
+    ignored_exceptions = os.getenv('THOTH_SENTRY_IGNORE_EXCEPTION')
+    if ignored_exceptions:
+        exceptions_split = ignored_exceptions.split(',')
+        if 'exc_info' in hint:
+            exc_type, exc_value, tb = hint['exc_info']
+            for exception in exceptions_split:
+                if exception == exc_type.__name__:
```

Comment:
Will this work if two libraries will implement an exception with a same name?

Let's say we will use two libraries `lib1` and `lib2`. Both implement an exception called `MyException` in `lib1.exceptions.MyException` and `lib2.exceptions.MyException`. Can we distinguish these two exceptions?

Reply:
I think we would need to add `lib1.exceptions.MyException,lib2.exceptions.MyException`, both to the env variables to ignore them. Alternatively we could check if MyException is in the string `lib1.exceptions.MyException`, but that might ignore more errors than we want to I guess.

Reply:
Is that what you were asking?

Reply:
> We would need to add `lib1.exceptions.MyException,lib2.exceptions.MyException`, both to the env variables to ignore them.

What if we want to ignore just `MyException` from `lib1`, but keep reporting `MyException` from `lib2`. Will it be possible?

Reply:
Yes, it should be, as here for example, we are checking for 'connexion.decorators.validation', this doesn't ignore 'xyz.validation'. 
However for generic python errors like ValueError, we are checking for exc_info type, not value, that is here in this example class type - 
`{'exc_info': (<class 'ValueError'>, ValueError('My really new error',), <traceback object at 0x7f2938edd0c8>), 'log_record': <LogRecord: test, 40, app.py, 10, "exception raised">}`
Should we catch the value? Btw that would only prevent handled errors, and the specific messages need to be added to env variables, which doesn't seem like a good idea.

Reply:
No exception value needed - we need to capture the correct exception type from the right module.

Could you verify this works as expected with your patch?

```console
$ cat lib1.py                                                                
class Test(Exception):
    pass
                                                                                
$ cat lib2.py                                                                
class Test(Exception):
    pass
                                                                                
$ cat test.py                                                                
import sys

from lib1 import Test as Test1
from lib2 import Test as Test2

try:
    raise Test1
except Exception:
    print(sys.exc_info()[0].__name__)
                                                                                
$ python3 test.py                                                            
Test
```

Reply:
So, the str(exc_type) returns - `<class 'lib1.Test'> `
Where as `exc_type.__name__`  returns 'Test' i.e. doesn't carry which module it is from. 
Should we do str(exc_type) and then some sort of string formatting?

Reply:
```
print(sys.exc_info())

print(str(sys.exc_info()[0]))

print(sys.exc_info()[0].__name__)

print(sys.exc_info()[0].__qualname__)
```

Gives - 
````
(<class 'lib1.Test'>, Test(), <traceback object at 0x7f3fc9fa0ec8>)

<class 'lib1.Test'>

Test

Test

Reply:
No string passing/ formatting needed. Can you come up with some other solution? 

Hint: `getattr` and `str.rsplit`

Reply:
The two methods above are your friends 🙂

Reply:
Or simply `inspect.getmodule`.

Reply:
Something like this?
`print(getattr(exc_type, "__module__")+'.'+exc_type.__name__)`
`lib1.Test`
These are all the attributes exc_type class has - 
`{'__module__': 'lib1', '__weakref__': <attribute '__weakref__' of 'Test' objects>, '__doc__': None}
`
Doesn't str.rsplit fall under string formatting? :stuck_out_tongue:

Reply:
Does this make sense?
`print(getattr(exc_type, "__module__")+'.'+exc_type.__name__)`
inspect.getmodule gives (plus we need to import inspect) - 
`<module 'lib1' from '/home/sai/Downloads/thoth-station/common/lib1.py'>
`
These are all the attr of exc_type - 
`{'__module__': 'lib1', '__weakref__': <attribute '__weakref__' of 'Test' objects>, '__doc__': None}
`

Reply:
But with this approach for errors like ValueError, we need to specify `builtins.ValueError` in the env variable.

Reply:
> Something like this?
> `print(getattr(exc_type, "__module__")+'.'+exc_type.__name__)`
> `lib1.Test`
> These are all the attributes exc_type class has -
> `{'__module__': 'lib1', '__weakref__': <attribute '__weakref__' of 'Test' objects>, '__doc__': None} `

Using the `__module__` attribute directly with `__name__` sounds good to me. :+1: 

> Doesn't str.rsplit fall under string formatting?

I don't this so: https://docs.python.org/3/library/string.html